### DISCUSSION TITLE
Custom columns

### DISCUSSION THREAD

**Discussion by micronc** [nickcoutsos/keymap-editor#257] [Ideas]
With all the different boards out there having a way to remove a column would be really great.

**Comment by nickcoutsos** [comment]
If you're talking about boards like the Corne that have 5 or 6 column variants you'd need to tell ZMK which you're using via the chosen node: https://zmk.dev/docs/config/kscan#example-skipping-unused-positions

If Keymap Editor knows the keyboard it will consider such a setting if it's found while parsing your keymap file (not in any other file) so borrowing Corne as an example again, if you added the following to `config/corne.keymap`:

```dts
/ {
  chosen {
    zmk,matrix-transform = &five_column_transform;
  };
};
```

then the app will render the correct layout.

**Reply by aaronm94** [reply]
Hi Nick, I tried doing this, and the keymap editor is showing 5 layers, but when I flash the firmware onto my 5 column corne, the numbers shift 1 over 

![image_2024-11-04_170826029](https://github.com/user-attachments/assets/4e587dda-9330-4e4d-bd80-ec6c79503687)

Do you have any idea why this might be happening? And how I could fix it? Thank you

**Reply by nickcoutsos** [reply]
Hi @aaronm94 , it's hard to say. If you have a repo you can share that would be the best way to pinpoint the issue. 

One thing I'd want to rule out is whether a different matrix transform is being selected elsewhere in the config that could office the keymap file.

To clarify, in your diagram, are you saying that pressing Q _results_ in W, or that _getting_ Q means pressing W?

**Reply by aaronm94** [reply]
I have been working on a fix,

https://github.com/aaronm94/corner

escentially, it is this same code except I added this line of code like  you said:

/ {
  chosen {
    zmk,matrix-transform = &five_column_transform;
  };
};

The issue is that the Top left button isn't being registered as anything and is shifting everything to the right, so the q would be in the place of the W in the image and nothing would be in the q's place. Another example is, in place of the A the P would be there.

**Reply by nickcoutsos** [reply]
That's strange... it doesn't look like there's anything unusual about your configuration. Maybe the first thing I'd try is changing the keys to something very different (e.g. "A B C" instead of "Q W E") and flash that to make sure the firmware is in fact updating.

If that's not it... maybe try and rule out a hardware error? I don't know if this is a handwired board but the off-by-one makes me think there's something unusual there. You might want to reach out to the vendor for help.

**Reply by nickzou** [reply]
> I have been working on a fix,
> 
> https://github.com/aaronm94/corner
> 
> escentially, it is this same code except I added this line of code like you said:
> 
> / { chosen { zmk,matrix-transform = &five_column_transform; }; };
> 
> The issue is that the Top left button isn't being registered as anything and is shifting everything to the right, so the q would be in the place of the W in the image and nothing would be in the q's place. Another example is, in place of the A the P would be there.

I'm also encountering this issue. Did you ever find a fix?

**Reply by aaronm94** [reply]
> > I have been working on a fix,
> > https://github.com/aaronm94/corner
> > escentially, it is this same code except I added this line of code like you said:
> > / { chosen { zmk,matrix-transform = &five_column_transform; }; };
> > The issue is that the Top left button isn't being registered as anything and is shifting everything to the right, so the q would be in the place of the W in the image and nothing would be in the q's place. Another example is, in place of the A the P would be there.
> 
> I'm also encountering this issue. Did you ever find a fix?

I figured out that I had a temper board instead of a corne. Lol.

**Comment by micronc** [comment]
This is what I am referring to. Since I am new to this world of ZMK and
programming keyboards simple things like this are taking me forever to
learn because I'm reading more on things that don't directly relate to what
I am looking for to get to what I am looking for. I've been using the key
mapping UI as a way of learning.

On Mon, Sep 23, 2024 at 12:07 AM Nick Coutsos ***@***.***>
wrote:

> If you're talking about boards like the Corne that have 5 or 6 column
> variants you'd need to tell ZMK which you're using via the chosen node:
> https://zmk.dev/docs/config/kscan#example-skipping-unused-positions
>
> If Keymap Editor knows the keyboard it will consider such a setting if
> it's found while parsing your keymap file (not in any other file) so
> borrowing Corne as an example again, if you added the following to
> config/corne.keymap:
>
> / {
>   chosen {
>     zmk,matrix-transform = &five_column_transform;
>   };
> };
>
> then the app will render the correct layout.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nickcoutsos/keymap-editor/discussions/257#discussioncomment-10721769>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMLPJIWPQ34DXGTSMGEQO3ZX6HZDAVCNFSM6AAAAABOVETLNWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANZSGE3TMOI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Regards,

Khalil R. Gonsalves
